core/src/main/java/com/linecorp/armeria/client/AbstractClientOptionsBuilder.java

```diff
@@ -294,6 +303,55 @@ public AbstractClientOptionsBuilder setHttpHeaders(
         return this;
     }
 
+    /**
+     * Sets the
+     * <a href="https://en.wikipedia.org/wiki/Basic_access_authentication">HTTP basic access authentication</a>
+     * header using {@link StandardCharsets#UTF_8}.
+     */
+    public AbstractClientOptionsBuilder basicAuth(String username, String password) {
+        return basicAuth(username, password, StandardCharsets.UTF_8);
+    }
+
+    /**
+     * Sets the
+     * <a href="https://en.wikipedia.org/wiki/Basic_access_authentication">HTTP basic access authentication</a>
+     * header. The specified {@link Charset} must be compatible with US-ASCII.
+     */
+    public AbstractClientOptionsBuilder basicAuth(String username, String password, Charset charset) {
+        requireNonNull(username, "username");
+        requireNonNull(password, "password");
+        requireNonNull(charset, "charset");
+        checkState(!oAuth1aSet || !oAuth2Set, "cannot set basic auth with %s.", oAuth1aSet ? "OAuth 1.0"
+                                                                                           : "OAuth 2.0");
+        httpHeaders.set(HttpHeaderNames.AUTHORIZATION,
+                        "Basic " + Base64.getEncoder().encodeToString(
+                                (username + ':' + password).getBytes(charset)));
+        return this;
+    }
+
+    /**
+     * Sets the <a href="https://oauth.net/core/1.0a/">OAuth Core 1.0 Revision A</a> header.
+     */
+    public AbstractClientOptionsBuilder oAuth1a(OAuth1aToken token) {
+        requireNonNull(token, "token");
+        checkState(!basicAuthSet || !oAuth2Set, "cannot set OAuth 1.0 with %s.", basicAuthSet ? "basic auth"
+                                                                                              : "OAuth 2.0");
+        httpHeaders.set(HttpHeaderNames.AUTHORIZATION, token.toHeaderValueString());
+        return this;
+    }
+
+    /**
+     * Sets the <a href="https://www.oauth.com/">OAuth 2.0</a> header. Note that the specified
+     * {@code accessToken} is sent using {@link HttpHeaderNames#AUTHORIZATION} header, not post body parameters.
+     */
+    public AbstractClientOptionsBuilder oAuth2(String accessToken) {
```

Comment:
- How about providing the overloaded methods that accept `BasicToken` and `OAuth2Token`?
- .. or just `auth(BasicToken|OAuth1aToken|OAuth2Token)`, so that an IDE gives the full choices when he or she types '.a`?
  ```java
  builder.auth(BasicToken.of(...));
  builder.auth(OAuth2Token.of(...));
  ```

Reply:
> builder.auth(BasicToken.of(...));
builder.auth(OAuth2Token.of(...));

Thanks! I like this idea. 😉

Reply:
Fixed. 😉